File: cpp/02_even_fibonacci_numbers.cpp
#include <iostream>

int even_fibonacci_numbers(int max_num){
    int sum_nums = 0;
    int fib[2] = {0, 1};
    
    while (fib[1] < max_num){
        int temp = fib[0];
        fib[0] = fib[1];
        fib[1] += temp;
        if (fib[1] % 2 == 0){
            sum_nums += fib[1];
        }
    }
    return sum_nums;
}

int main(){
    std::cout << even_fibonacci_numbers(4'000'000) << "\n";
}

File: c#/_02_EvenFibonacciNumbers.cs
using System;

public class _02_EvenFibonacciNumbers{
    public static int euler02(){
        int n1 = 1, n2 = 0, total = 0;
        while (n2 < 4_000_000){
            int temp = n2;
            n2 += n1;
            n1 = temp;
            if (n2 % 2 == 0) total += n2;
        }
        return total;
    }
    
    public static void Main(string[] args){
        Console.WriteLine(euler02());
    }
}
